### PR TITLE
Added a few samples programs for TypeScript and JavaScript (W/docs)

### PR DESCRIPTION
**Pull request by Ray6464**
Added the following programs:
1. Capitalize for TypeScript
2. Reversing a string for TypeScript
3. Import/export for JavaScript
4. Import/Export for TypeScript

Also added documentation for import/export, with articles for JavaScript, and TypeScript.

- [x] I fixed ##1179 & ##1178

### Code Snippets
- [x] I created/updated the language README
  - [x ] I added the sample program name to the README

### Documentation
- [x] I followed the applicable article template
  - [x] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [x] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [x] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

## Notes
Added the following programs:
1. Capitalize for TypeScript
2. Reversing a string for TypeScript
3. Import/export for JavaScript
4. Import/Export for TypeScript
Also added documentation for import/export, with articles for JavaScript, and TypeScript.

### CODE REVIEW

File: docs/projects/import-export/_posts/2019-10-01-javascript.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: importing and exporting a variable in JavaScript  
+layout: default  
+last-modified: 2019-10-01  
+featured-image:  
+tags: [javascript, import, export]  
+authors:  
+  - ray6464
```
Comment (jrg94) [inline docs/projects/import-export/_posts/2019-10-01-javascript.md]: Have you added yourself to [the authors file](https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/_data/authors.yml)?

File: docs/projects/import-export/_posts/2019-10-01-javascript.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+title: importing and exporting a variable in JavaScript  
+layout: default  
+last-modified: 2019-10-01
```
Comment (jrg94) [inline docs/projects/import-export/_posts/2019-10-01-javascript.md]: You can probably update the date here (as well as in the title). Same goes for the other articles.

File: docs/projects/import-export/_posts/2019-10-01-javascript.md (hunk 1 above)
Comment (Ray6464) [reply docs/projects/import-export/_posts/2019-10-01-javascript.md]: Should I add myself to the file on the following link?
https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/_data/authors.yml

File: docs/projects/import-export/_posts/2019-10-01-javascript.md (hunk 2 above)
Comment (Ray6464) [reply docs/projects/import-export/_posts/2019-10-01-javascript.md]: Fixed dates!

File: docs/projects/import-export/_posts/2019-10-01-javascript.md (hunk 1 above)
Comment (Ray6464) [reply docs/projects/import-export/_posts/2019-10-01-javascript.md]: Added myself to author's file

File: docs/_data/authors.yml [hunk 3]
```diff
@@ -130,6 +130,14 @@ veevidify:
   web:
   bio:
 
+
+ray6464:
+  name: Ray Voice
+  email: ray@vanet@gmail.com
+  web: twitter: @rayshorthead
```
Comment (jrg94) [inline docs/_data/authors.yml]: Looks like this syntax breaks the YAML. Could you link the Twitter URL here?

Comment (Ray6464) [reply docs/_data/authors.yml]: fixed

**Pull request merged**